jax/interpreters/xla.py

```diff
@@ -190,6 +190,17 @@ def _check_nans(name, xla_shape, buf):
         msg = "invalid value (nan) encountered in {}"
         raise FloatingPointError(msg.format(name))
 
+def _make_xla_args(c, *xla_shapes):
+  all_shapes, treedef = tree_util.tree_flatten(xla_shapes)
```

Comment:
Why flatten the shapes? I'm not sure they ever have nested structure to them in the first place, but even if they did, you wouldn't need to alter it here. You just want to wrap a tuple around the whole thing.

Reply:
This was to support my complicated, undocumented _make_xla_args API where you pass in a list of lists of shapes, and then it returns a list of list of XLA values of the same shape. This was to make it easy to call in jaxpr_computation. I got rid of this function and inlined the logic instead though, since I'm only calling it in jaxpr_computation now.